code/controlconfig/controlsconfigcommon.cpp

```diff
@@ -1517,6 +1583,25 @@ void control_config_common_read_section(int s) {
 			if (optional_string("$Disable:")) {
 				stuff_boolean(&item->disabled);
 			}
+
+			if (optional_string("+Locked")) {
+				item->locked = true;
+			}
+
+			if (optional_string("Locked:")) {
+				stuff_boolean(&item->locked);
+			}
+
+			if (optional_string("$OnActionHook:")) {
+				SCP_string buf;
+				sprintf(buf, "%s - %s", "controlconfigdefault.tbl", item->text.c_str());
+
+				auto* hook = &Lua_hooks[static_cast<IoActionId>(item_id)];
+				Script_system.ParseChunk(&hook->second, buf.c_str());
+
+				item->scriptEnabledByDefault = optional_string("+EnableHookByDefault");
+				hook->first = item->scriptEnabledByDefault;
+			}
```

Comment:
We generally do not do include Lua code in other tables since it can lead to some very weird situations and generally is not as flexible.

Instead, the usual way to do this is to expose an API function where a script can register a function to be called when something relevant happens. See the `LuaSEXP` code for some examples.

Reply:
This was done to keep consistent with the Hook: [] +Override: [] syntax used for other types of hooks, as these work semantically identical (with hook being executed every time and the optional override disabling usual in-engine handling).
In addition, you shouldn't really change your hook callback mid-game (just like all the other hooks).
If it is still desired to break that analogy for keeping all lua in sct-tables, I'd change it to a function registerHook(fnc hook, bool enabledByDefault = false, bool isOverride = false).
Also, while this is not the main repo, since it is going to be part of the control overhaul, it needs to conform with main-repo standarts anyways.

Reply:
A big part about not having Lua code in tables is that it basically forces a script to keep some global data or else it will not be possible to interact between the "main" script (if such a thing exists) and the controls code.

The proposed signature you posted looks good to me at least. If you ware worried about changes mid-mission then it is perfectly acceptable to just throw an error when that is done in any other place than the `On Game Init` hook.